seaworthy/containers/redis.py

```diff
@@ -0,0 +1,44 @@
+from seaworthy.definitions import ContainerDefinition
+from seaworthy.logs import output_lines
+
+
+class RedisContainer(ContainerDefinition):
+    DEFAULT_NAME = 'redis'
+    DEFAULT_IMAGE = 'redis:alpine'
+    DEFAULT_WAIT_PATTERNS = (r'\* Ready to accept connections',)
+
+    def __init__(self,
+                 name=DEFAULT_NAME,
+                 image=DEFAULT_IMAGE,
+                 wait_patterns=DEFAULT_WAIT_PATTERNS,
+                 **kwargs):
+        super().__init__(name, image, wait_patterns, **kwargs)
+
+    def base_kwargs(self):
+        return {'tmpfs': {'/data': 'uid=100,gid=101'}}
+
+    def clean(self):
+        self.exec_redis_cli('FLUSHALL')
+
+    def exec_redis_cli(self, command, args=[], db=0, cli_opts=[]):
+        """
+        Execute a ``redis-cli`` command inside a running container.
+
+        :param command: the command to run
+        :param args: a list of args for the command
+        :param db: the db number to query (default ``0``)
+        :param cli_opts: a list of extra options to pass to ``redis-cli``
+        """
+        cli_opts = ['-n', str(db)] + cli_opts
+        cmd = ['redis-cli'] + cli_opts + [command] + [str(a) for a in args]
```

Comment:
Should we standardise across containers on:
* Argument names, e.g. RabbitMQ has `command_opts` and `rabbitmqctl_opts`, whereas here we have `args` and `cli_opts`
* Stringifying: `[str(a) for a in args]`, which isn't done in the RabbitMQ container.
?

I think I kind of prefer the way you've done things here, and maybe we should standardise on `args`/`cli_opts`. Just don't want weird inconsistencies.

Reply:
I think `args`/`<name of command>_opts` is the best option, but `redis_cli_opts` was a bit long. :-/

The stringifying isn't really necessary for `rabbitmqctl`, because all the args are already strings. For `redis-cli`, we're much more likely to be passing ints and such. I'm happy to stringify by default, though.